### PR TITLE
[stable/elasticsearch-exporter] Add support for servicemonitor (prometheus operator)

### PR DESCRIPTION
**Pull request by fabriziofortino**
#### What this PR does / why we need it:
Add support for prometheus operator `ServiceMonitor`.

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by fabriziofortino** [comment]
@sameersbn @foxish @stealthybox I think the `ServiceMonitor` resource should live in this chart since it's strictly related to the exporter. Could you please review? 🙇

File: stable/elasticsearch-exporter/templates/servicemonitor.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+{{ if .Values.serviceMonitor.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: {{ include "elasticsearch-exporter.fullname" . }}
+  {{- if .Values.serviceMonitor.namespace }}
+  namespace: {{ .Values.serviceMonitor.namespace }}
+  {{- end }}
+  labels:
+    {{- range $key, $value := .Values.serviceMonitor.selector }}
```
Comment (stealthybox) [inline stable/elasticsearch-exporter/templates/servicemonitor.yaml]: I recommend changing this from `serviceMonitor.selector` to `serviceMonitor.labels`.
The label selector is defined in the operator config.

**Review by stealthybox — CHANGES_REQUESTED** [review]
In general, I support getting this change merged.
I think we should also have the capability in the main Elasticsearch chart.

File: stable/elasticsearch-exporter/templates/servicemonitor.yaml (hunk 1 above)
Comment (fabriziofortino) [reply stable/elasticsearch-exporter/templates/servicemonitor.yaml]: done

**Comment by fabriziofortino** [comment]
@stealthybox my original idea was to integrate the `exporter` and the `servicemonitor` in the main ES chart (as proposed here https://github.com/helm/charts/pull/8179). The charts maintainers decided to put the exporter as a requirement in the `elastic-stack` chart and leave the main elasticsearch chart unchanged. I think it makes sense to have the servicemonitor as part of the exporter.

**Comment by stealthybox** [comment]
> I think it makes sense to have the servicemonitor as part of the exporter.

I agree

> The charts maintainers decided to <...> leave the main elasticsearch chart unchanged

Yes -- I disagree with this, but it's unrelated to the quality of your change.

_______________
Chart.yaml looks to need a version bump, otherwise:
/lgtm

**Pull request closed without merging**